test/e2e_node/podresources_test.go

```diff
@@ -702,6 +702,26 @@ var _ = SIGDescribe("POD Resources [Serial] [Feature:PodResources][NodeFeature:P
 					podresourcesListTests(ctx, f, cli, nil)
 					podresourcesGetAllocatableResourcesTests(ctx, cli, nil, onlineCPUs, reservedSystemCPUs)
 				})
+				ginkgo.It("should accounts resources for pods in terminal phase", func(ctx context.Context) {
+					pd := podDesc{
+						cntName:    "e2e-test-cnt",
+						podName:    "e2e-test-pod",
+						cpuRequest: 1000,
+					}
+					pod := makePodResourcesTestPod(pd)
+					pod.Spec.Containers[0].Command = []string{"sh", "-c", "sleep 10s"}
+					pod = e2epod.NewPodClient(f).CreateSync(ctx, pod)
+					err := e2epod.WaitForPodCondition(ctx, f.ClientSet, pod.Namespace, pod.Name, "Pod Succeeded", time.Minute*2, testutils.PodSucceeded)
+					framework.ExpectNoError(err)
+					endpoint, err := util.LocalEndpoint(defaultPodResourcesPath, podresources.Socket)
+					framework.ExpectNoError(err)
+					cli, conn, err := podresources.GetV1Client(endpoint, defaultPodResourcesTimeout, defaultPodResourcesMaxSize)
+					framework.ExpectNoError(err)
+					defer conn.Close()
+					// although the pod moved into terminated state, PodResourcesAPI still list its cpus
```

Comment:
How is the pod in terminated state here? Call to list endpoint is made prior to pod deletion so it makes sense for cpus to be reported here. It looks like we need an additional check for the resources reported by podresource endpoint while the pod is being in deleted and thus in terminal phase.

Reply:
According to the docs terminal phase is a pod which reports of `Failed` or `Succeeded` status.
In line 714 we wait for the pod to transition into `Succeeded` and only then we call `List` via the podres client.

Reply:
Thanks for clarifying. I mixed up `terminal` and `terminated`.